storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -705,9 +705,20 @@ bool xbcloud_put(Object_store *store, const std::string &container,
     return false;
   }
 
-  // check if xtrabackup_info, almost the last file exists in backup
-  if (std::count(object_list.begin(), object_list.end(),
-                 backup_name + "/xtrabackup_info.00000000000000000000") == 0) {
+  /* check if the last_file (xtrabackup_tablespaces.gz.00000000000 or
+  xtrabackup_tablespaces.00000000000000000000) is uploaded to cloud storage to
+  determine successful xbcloud "put" operation. */
+  std::string last_file_prefix = backup_name + "/xtrabackup_tablespaces";
```

Comment:
I believe there is confusion here. xtrabackup_tablespaces is not a valid file on 2.4. It's only available on 8.0 branch at storage/innobase/xtrabackup/src/space_map.cc .

Reply:
thank you @altmannmarcelo for pointing this out. right. this is not a valid file. 

> ./run.sh -t t/xbcloud.sh
> Using /home/rahul.malik/MySQL/build/x7/xtrabackup-test/var as test root
> Detecting server version...
> Running against MySQL 5.7.31-log (InnoDB 5.7.31)
> Using 'xtrabackup' as xtrabackup binary
> Using 8 parallel workers
> 
> ## ==============================================================================
> TEST WORKER RESULT TIME(s) or COMMENT
> # xbcloud w1 [passed] 72
> Spent 72 of 79 seconds executing testcases
> 
> SUMMARY: 1 run, 1 successful, 0 skipped, 0 failed
> 
> See results/ for detailed output
> {smm-12-d40aec x7 o7 xtrabackup-test}$

This result is wrong. Mistakely I executed the test on wrong build.

Reply:
What is the correct file then?

This is the reason local results cannot be trusted :cry:

Reply:
incase of 2.4 last file can be xtrabackup_info or xtrabackup_log. so instead of relying upon the last file I thinking to  iterate through list xtrabackup_info file is present